common/service.go

```diff
@@ -19,6 +19,8 @@ type Service interface {
 	IsRunning() bool
 
 	String() string
+
+	SetLogger(log.Logger)
```

Comment:
Can u remind me why the constructors are taking loggers if we have this method ?